### PR TITLE
Api logic divide

### PR DESCRIPTION
**Pull request by Ilya-Mir**
- Separate api logic
- Partly fixed linter error

### CODE REVIEW

File: src/generic/api.js [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+const responseSchemaHandler = resp => resp;
+
+export class ApiAuth {
+  constructor(web3) {
+    this.web3 = web3;
+    this.logIn = this.logIn.bind(this);
+    this.addName = this.addName.bind(this);
+  }
+
+  logIn = () => {
```
Comment (rumkin) [inline src/generic/api.js]:
```
logIn = () => {
}
```
is the same as
```
logIn() {
}
```

File: src/generic/api.js [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+
+  fetch = () =>
+    apiGet('/time', null)
+      .then(errorHandler)
+      .then(responseSchemaHandler);
+}
+
+export class ApiKeys {
+  constructor() {
+    this.fetch = this.fetch.bind(this);
```
Comment (rumkin) [inline src/generic/api.js]: Useles context  binding if `this.fetch` is arrow function.

File: src/actions/offers.js [hunk 3]
```diff
@@ -10,10 +10,11 @@ export const VERIFY_OFFER = 'VERIFY_OFFER';
 export const PAY_OFFER = 'PAY_OFFER';
 export const NEW_OFFER = 'NEW_OFFER';
 
+const OffersApi = new ApiOffers();
```
Comment (goblinrd) [inline src/actions/offers.js]: web3 not injected

File: src/generic/api.js [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+          web3.eth.getAccounts((err, accs) => {
+            if (err) {
+              alert('Metamask error: cannot get account');
+            } else {
+              const account = accs[0];
+              if (!account) {
+                alert('Unlock metamask');
+                return;
+              }
+              res(this.sendTransaction(CONTRACT_ADDRESS, offer, ETHEREUM_NET));
```
Comment (goblinrd) [inline src/generic/api.js]: static method invocation as instance method

**Pull request merged**